comps/llms/text-generation/predictionguard/predictionguard_llm.py

```diff
@@ -1,64 +1,57 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+# TODO: Add streaming
 import os
 from fastapi.responses import StreamingResponse
-from langsmith import traceable
-from comps import GeneratedDoc, LLMParamsDoc, ServiceType, opea_microservices, register_microservice
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
 from predictionguard import PredictionGuard
-import os
-import requests
-from langsmith import traceable
+from comps import GeneratedDoc, ServiceType, opea_microservices, register_microservice
+
+# Set up Prediction Guard API key
+PG_API_KEY = os.getenv("PREDICTIONGUARD_API_KEY")
+
+# Define the input data model
+class LLMParamsDoc(BaseModel):
+    model: str = "Neural-Chat-7B"
+    query: str
+    max_new_tokens: int
+    temperature: float
+    top_p: float
+    top_k: int
+    #stream: bool = False
 
-# Initialize Prediction Guard API URL and Key
-PG_API_URL = "https://api.predictionguard.com/completions"
-PG_API_KEY = os.getenv("PREDICTION_GUARD_API_KEY")
+app = FastAPI()
 
 @register_microservice(
     name="opea_service@llm_predictionguard",
     service_type=ServiceType.LLM,
-    endpoint="/v1/chat/completions",
+    endpoint="/v1/completions",
     host="0.0.0.0",
     port=9000,
 )
-@traceable(run_type="llm")
 def llm_generate(input: LLMParamsDoc):
-    headers = {
-        "Authorization": f"Bearer {PG_API_KEY}",
-        "Content-Type": "application/json"
-    }
+    if not PG_API_KEY:
+        raise HTTPException(status_code=500, detail="PREDICTIONGUARD_API_KEY environment variable is not set")
 
-    data = {
-        "model": "Neural-Chat-7B",
-        "prompt": input.query,
-        "max_tokens": input.max_new_tokens,
-        "temperature": input.temperature,
-        "top_p": input.top_p,
-        "top_k": input.top_k,
-        "streaming": input.stream
-    }
-    # TODO: figure out how this works with streaming
-    # if input.streaming:
-    #     async def stream_generator():
-    #         response = requests.post(PG_API_URL, headers=headers, json=data, stream=True)
-    #         chat_response = ""
-    #         for line in response.iter_lines():
-    #             if line:
-    #                 text = line.decode('utf-8')
-    #                 chat_response += text
-    #                 chunk_repr = repr(text.encode("utf-8"))
-    #                 print(f"[llm - chat_stream] chunk:{chunk_repr}")
-    #                 yield f"data: {chunk_repr}\n\n"
-    #         print(f"[llm - chat_stream] stream response: {chat_response}")
-    #         yield "data: [DONE]\n\n"
+    client = PredictionGuard()
```

Comment:
I think we should instantiate the client globally instead of on every request. That also means the service will fail to start if the environment variable is not set which is good.